clang/lib/Driver/ToolChains/Cuda.cpp

```diff
@@ -755,6 +755,11 @@ void CudaToolChain::addClangTargetOptions(
     if (DriverArgs.hasFlag(options::OPT_fcuda_approx_transcendentals,
                            options::OPT_fno_cuda_approx_transcendentals, false))
       CC1Args.push_back("-fcuda-approx-transcendentals");
+
+    if (DriverArgs.hasArg(options::OPT_fsycl)) {
+      // Add these flags for .cu SYCL compilation.
+      CC1Args.append({"-std=c++17", "-fsycl-is-host"});
```

Comment:
Is there any concern of overriding the `-std` version via the command line?

Reply:
Hi @mdtoguchi, thanks for your review.

A different `-std` specified in the compilation command has the precedence over the `-std=c++17` introduced above.
This is due to the fact that the `-std` introduced by the compilation command is append after `-std=c++17`.